tests/PyPI-pytest/run_all.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+#     Copyright 2019, Kay Hayen, mailto:kay.hayen@gmail.com
+#
+#     Python test originally created or extracted from other peoples work. The
+#     parts from me are licensed as below. It is at least Free Software where
+#     it's copied from other people. In these cases, that will normally be
+#     indicated.
+#
+#     Licensed under the Apache License, Version 2.0 (the "License");
+#     you may not use this file except in compliance with the License.
+#     You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#     Unless required by applicable law or agreed to in writing, software
+#     distributed under the License is distributed on an "AS IS" BASIS,
+#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#     See the License for the specific language governing permissions and
+#     limitations under the License.
+#
+
+import os
+import sys
+import subprocess
+
+from nuitka.tools.testing.OutputComparison import compareOutput
+from nuitka.tools.testing.Virtualenv import withVirtualenv
+from nuitka.utils.FileOperations import removeDirectory
+
+packages = (
+    "urllib3",
+)
+
+base_dir = os.getcwd()
+
+
+
+
+for package_name in packages:
+    try:
+        # get normal pytest results
+        with withVirtualenv("venv_%s" % package_name) as venv:
+            venv.runCommand(
+                commands=[
+                "call Scripts/activate",
+                "git clone https://github.com/%s/%s.git" % (package_name,package_name),
+                "python -m pip install pytest",
+                "cd %s" % package_name,
+                "python -m pip install -r dev-requirements.txt",
+                # "python -m pytest --disable-warnings",
+
+                #"cd ..",
+                #"git clone https://github.com/nuitka/nuitka.git",
+                #"cd nuitka",
+                #"python setup.py develop",
+                #"cd ../%s" % package_name,
+                #"python setup.py bdist_nuitka",
+                # "python -m pip install dist/%s" % os.listdir(os.path.join(base_dir, "venv2_%s" % package_name, package_name, "dist"))[0],
+
+                ]
+            )
+
+
+            # need to use subprocess to save stdout
+            # currently doesn't work due to import errors
+
+            os.chdir(os.path.join(base_dir, "venv_%s" % package_name, package_name))
+            process = subprocess.Popen(
+                    args="python -m pytest --disable-warnings",
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                )
+            stdout, stderr = process.communicate()
```

Comment:
Using subprocess I can redirect stdout and save it as a string, this is also what `distutils/run_all.py` does. However, pytest gets import errors when ran with subprocesss

Reply:
Is it maybe because the virtualenv isn't activated here?